main.tf

```
@@ -223,7 +223,7 @@ resource "aws_security_group_rule" "allow_icmp_ingress" {
 }
 
 resource "aws_security_group_rule" "alb" {
-  count                    = var.enabled ? 1 : 0
+  count                    = var.enabled && var.alb_security_group == "" ? 1 : 0
```

I think it should be:

```suggestion
  count                    = var.enabled && var.alb_security_group != "" ? 1 : 0
```

---

I also think then variable: `alb_security_group` could be optional and maybe renamed to something like `default_ingress_security_group`

---

agree
@jamengual please address the `count` comment

---

will do, sorry It was to late last night did not realize I pushed the wrong commit

---

so should I rename the vars too? like : 
`resource "aws_security_group_rule" "default" {
  count                    = var.enabled && var.default_ingress_security_group_id != "" ? 1 : 0
  type                     = "ingress"
  from_port                = 0
  to_port                  = var.container_port
  protocol                 = "tcp"
  source_security_group_id = var.default_ingress_security_group_id
  security_group_id        = join("", aws_security_group.ecs_service.*.id)
}`

---

@jamengual 
let's keep the old name for now to not change the module interface that can affect all other modules that uses it

---

ok

---

done